{{cookiecutter.github_repository}}/{{cookiecutter.main_module}}/users/auth/serializers.py

```diff
@@ -43,3 +38,70 @@ def get_name(self, obj):
 
     def get_auth_token(self, obj):
         return get_token_for_user(obj, "authentication")
+
+
+class PasswordChangeSerializer(serializers.Serializer):
+    current_password = serializers.CharField(required=True)
+    new_password = serializers.CharField(required=True)
+
+    error_messages = {
+        'invalid_password': 'Invalid password.'
+    }
+
+    def validate_current_password(self, value):
+        if not self.context['request'].user.check_password(value):
+            raise serializers.ValidationError(self.error_messages['invalid_password'])
+        return value
+
+    def validate_new_password(self, value):
+        # https://docs.djangoproject.com/en/2.0/topics/auth/passwords/#django.contrib.auth.password_validation.validate_password
+        password_validation.validate_password(value)
+        return value
+
+
+class PasswordResetSerializer(serializers.Serializer):
+    email = serializers.EmailField(required=True)
+
+    error_messages = {
+        'email_does_not_exist': "No user with the specified email found"
```

Comment:
There shouldn't be any error message in this case. The response will be:
`Further instructions will be sent to the email if it exists`
or something similar

Reply:
@saurabh-fueled @theskumar Like we have already discussed, anyone can hit the registration api to gather information about the user-emails registered with the app. Should I leave this as is or make changes according to the comments above ?

Reply:
There are few things to consider though.
-  Error provides provides much better user experience in case of mail delays and typo.
- Will the registration feature always available, in an example of private invitation kind of setup. which in most of the cases won't be.

Considering these two I would give little more preference to security as they always tend to be forgotten by developer. As for having better UX, I would leave to it dev and implement the proper error as needed.